Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/callback/cxf/largefile/CONNECTTimestamp.java

```diff
@@ -89,38 +86,55 @@ public boolean isExpired(Date invocationDate) {
      * @return true if the timestamp is before (now-timeToLive), false otherwise
      */
     public boolean verifyCreated(int timeToLive, int futureTimeToLive, Date invocationDate) {
-        if (invocationDate == null) {
-            invocationDate = new Date();
-        }
+        Date date = invocationDate == null ? new Date() : invocationDate;
 
-        long invocationTime = invocationDate.getTime();
+        long invocationTime = date.getTime();
         if (futureTimeToLive > 0) {
-            invocationDate.setTime(invocationTime + futureTimeToLive * 1000);
+            date.setTime(invocationTime + futureTimeToLive * 1000);
```

Comment:
Nevermind.  The confusion is that the same variable is used to represent two values, the value of the future time (so it's not created in the future) and of the past time (so it's confirmed that it isn't created too far in the past.  We might want to clean this up for clarity and appropriate variable naming.  Using a variable name like "date" isn't very clear.